src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -155,13 +157,16 @@ public EnhancedSortedSet<TValue> GetFirsts()
         /// <summary>
         /// Gets last element in supplied comparer order.
         /// </summary>
-        [MethodImpl(MethodImplOptions.Synchronized)]
-        public bool TryGetLast(out TValue last)
+        public bool TryGetLast(out TValue? last)
         {
-            last = _worstSortedValues.Max.Key;
+            last = _worstValue is not null ? _worstValue.Value.Key : default;
```

Comment:
Risk of null ref by using field twice; either take to local, or change to `?.` and it will take locals
```suggestion
            last = _worstValue?.Value?.Key;
```